### PR TITLE
Setup repository for us *as* a GitHub Action

### PR DESCRIPTION
**Pull request by alerque**
A follow up to #27 for an official Docker image and ability to run this repository *as* a GitHub action to lint other projects in a single easy to use workflow step.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#syntax=docker/dockerfile:1.2
+
+FROM alpine:latest AS luacheck
+
+LABEL maintainer="Caleb Maclennan <caleb@alerque.com>"
+
+RUN apk add --no-cache dumb-init lua lua-argparse lua-filesystem
```
Comment (MartinBroers) [inline Dockerfile:7]: For readability I always prefer to list the intstallable items in a list, like this:

```suggestion
RUN apk add --no-cache \
    dumb-init \
    lua \
    lua-argparse \
    lua-filesystem
```
and to reduce the container size, maybe remove the apk cache as well?

```
     rm -rf "/var/cache/apk/"*
```

Comment (alerque) [reply Dockerfile:7]: You don't have to remove the apk cache at all if you tell it `--no-cache`. It does have a directory still, but it is empty. I checked in the built image.

**Comment by alerque** [comment]
I can take care of it. I'm on the road right now (and for a while) so it might be a bit delayed, but I think I can handle the merge all at once with these changes and any further tweaking I do built in. Thanks.

**Comment by alerque** [comment]
I've been putting this off because I didn't know how the release process was going to go. I think that's coming together, so time to get this ready. It will mean (briefly) that the README describes something that isn't workable because the container registry won't have a Docker build matching the *last* tag release.

**Pull request merged**